src/Collection/test.js

```diff
@@ -206,24 +206,4 @@ describe('creating new records', () => {
     // no throw inside action
     await database.action(() => tasks.create(noop))
   })
-  it('can destroy records permanently', async () => {
-    // should this even be tested here? Shouldn't it be tested on Model, without mocks?
-    const { tasks: collection, adapter } = mockDatabase()
-    adapter.batch = jest.fn()
-
-    const observer = jest.fn()
-    collection.changes.subscribe(observer)
-
-    const m1 = new MockTask(collection, { id: 'm1' })
-    collection._cache.add(m1)
-
-    await collection._destroyPermanently(m1)
-
-    // Check database delete, cache delete, observers update
-    expect(adapter.batch).toHaveBeenCalledTimes(1)
-    expect(adapter.batch).toHaveBeenCalledWith([['destroyPermanently', m1]])
-    expect(collection._cache.get('m1')).toBe(undefined)
```

Comment:
this looks like something that's missing from your refactored test

Reply:
Could you expand a bit on what test you want? Is it not covered by https://github.com/Nozbe/WatermelonDB/pull/294/files#diff-1c7bf3678f7379f801755def5d7c2facR156 ?